flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkBase.java

```diff
@@ -208,6 +222,13 @@ public void invoke(T value) throws Exception {
 		checkErrorAndRethrow();
 
 		elasticsearchSinkFunction.process(value, getRuntimeContext(), requestIndexer);
+
+		// if there is a connectivity failure, then retry
+		if (failureThrowable.get() != null &&
```

Comment:
I'm wondering whether or not checking the exception type would be enough for verifying the connectivity failure.